### PR TITLE
Move test data files to framework

### PR DESCRIPTION
**Pull request by davidhorstmann-arm**
Move `tests/data_files` to the `framework` submodule. Paired with Mbed-TLS/mbedtls-framework#18. 

Fixes Mbed-TLS/mbedtls-framework#17.

## PR checklist

Please tick as appropriate and edit the reasons (e.g.: "backport: not needed because this is a new feature")

- [x] **changelog** not required - internal change only
- [x] **3.6 backport** #9375 
- [x] **2.28 backport** not required - no framework in 2.28
- [x] **tests** not required - file motion only

### CODE REVIEW

**Review by ronald-cron-arm — CHANGES_REQUESTED** [review]
This looks almost good to me.
I have checked that the "Move some files to framework repository" commits in mbedtls and mbedtls-framework are both based on the same commit: https://github.com/Mbed-TLS/mbedtls/commit/2a674bd9ce4758dff0d18f4ac8b6da4419efc504. The one in mbedtls removes the tests/data_files/* files while the one in mbedtls-framework removes all files but the tests/data_files/* ones that are renamed (path change only), same 773 files removed/renamed in both cases and only data_files/* remaining in the mbedtls-framework PR.

**Comment by davidhorstmann-arm** [comment]
@ronald-cron-arm @tom-daubney-arm I'll wait for your reviews before I rebase, unless it seems better to you to go ahead?

**Comment by tom-daubney-arm** [comment]
> @ronald-cron-arm @tom-daubney-arm I'll wait for your reviews before I rebase, unless it seems better to you to go ahead?

Fine for me if you wait.

File: tests/scripts/generate_tls13_compat_tests.py
Comment (ronald-cron-arm) [inline tests/scripts/generate_tls13_compat_tests.py:24]: Rather
```suggestion
    'ecdsa_secp256r1_sha256': Certificate('$DATA_FILES_PATH/test-ca2.crt',
```
I'd say with DATA_FILES_PATH defined in ssl-opt.sh and then update tls13-compat.sh accordingly.

**Review by ronald-cron-arm — CHANGES_REQUESTED** [review]

**Review by ronald-cron-arm — APPROVED** [review]
LGTM but got some conflicts now.

**Review by tom-daubney-arm — DISMISSED** [review]
Sorry for taking so long on this one. It took me a minute to work out the workflow here and to verify that the two streams of work are based on the same commit. But I have verified that now and can see they are both based on [2a674bd9ce](https://github.com/Mbed-TLS/mbedtls/commit/2a674bd9ce4758dff0d18f4ac8b6da4419efc504) which also agrees with @ronald-cron-arm's review.

This all LGTM and will review again once the conflicts are merged.

**Comment by davidhorstmann-arm** [comment]
Rebased by doing the following actions:

* Recreate the file-move commit on top of latest `development` and make a new branch from it
* Cherry-pick path-updating commits from the existing branch onto the new branch:
    * There were conflicts in some test data files due to test dependency renames (`MBEDTLS_XYZ_C` to `PSA_WANT_XYZ`). These were fairly easy to resolve.
    * I skipped framework update commits as these caused conflicts
 * Add a framework update commit at the end that updates to the latest version of the framework PR.

**Review by ronald-cron-arm — DISMISSED** [review]
LGTM. The "Move some files to framework repository" commits in mbedtls and mbedtls-framework are now both based on commit: https://github.com/Mbed-TLS/mbedtls/commit/675e150b0f6080eccb43db8e3a3a55ade50dca82.

**Review by tom-daubney-arm — DISMISSED** [review]
LGTM! I too confirm that after the rebase the commits are based on the same commit, being  [675e150](https://github.com/Mbed-TLS/mbedtls/commit/675e150b0f6080eccb43db8e3a3a55ade50dca82).

**Review by davidhorstmann-arm — COMMENTED** [review]
Submodule update LGTM

**Review by ronald-cron-arm — APPROVED** [review]
LGTM with the approval of the last commit by @davidhorstmann-arm

**Review by tom-daubney-arm — APPROVED** [review]
Updates LGTM, thanks.

**Pull request merged**
